app/controllers/twitter_users_controller.rb

```diff
@@ -0,0 +1,79 @@
+class TwitterUsersController < ApplicationController
+  before_action :current_user
+
+  def subscribe
+    @screen_name = params[:screen_name]
+    if twitter_user_protected?
+      flash[:error] = "That user is protected; you cannot subscribe to their tweets."
+    else
+      if !twitter_user_exists?
+        # if twitter_user doesn't exist, create twitter_user and their tweets
+        create_twitter_user(@screen_name)
+        create_tweets(@twitter_user)
+      end
+      @twitter_user = TwitterUser.find_by(screen_name: @screen_name)
+      if @current_user.twitter_users.include?(@twitter_user)
+        #if current_user is already subscribed to this twitter_user, show error
+        flash[:error] = "You are already subscribed to this user."
+      else
+        # subscribe current_user to twitter_user
+        @current_user.twitter_users << @twitter_user
+      end
+    end
+    redirect_to :root
```

Comment:
This method is pretty bulky for a controller method. I would suggest breaking out single pieces of functionality into separate methods that are called within this method.